client/modules/Manage/Bookings/components/BookingAdm.js

```diff
@@ -0,0 +1,365 @@
+    return (
+      <FormGroup>
+          {this.renderLabel(field, field)} <br />
+          <ButtonDropdown isOpen={this.state.dropdownsOpen[field]} toggle={(e) => this.toggleDropDown(field, e)}>            
+            <DropdownToggle caret size="sm">{text}</DropdownToggle>
+            <DropdownMenu>
+              {opts}
+            </DropdownMenu>
+          </ButtonDropdown>          
+      </FormGroup>
+    );
+  }
+
+  renderCarDropDown(field, options) {
+    const opts = options.map(m => (
+        <DropdownItem key={m._id} onClick={(e) => this.ddOnSelect(field, m, e)}>{m.make} {m.model} {m.rego}</DropdownItem>
+      )
+    );
+    const text = this.state.booking[field] ? 
+    this.state.booking[field].make + " " + this.state.booking[field].model + " " + this.state.booking[field].rego : 
+    `Select ${this.labels[field]}`;
+    return (
+      <FormGroup>
+          {this.renderLabel(field, field)} <br />
+          <ButtonDropdown isOpen={this.state.dropdownsOpen[field]} toggle={(e) => this.toggleDropDown(field, e)}>            
+            <DropdownToggle caret size="sm">{text}</DropdownToggle>
+            <DropdownMenu>
+              {opts}
+            </DropdownMenu>
+          </ButtonDropdown>          
+      </FormGroup>
+    );
+  }
+
+  submit(evt) {
+    evt.preventDefault()
+    this.setLoading(true);
+
+    manageSvc.bookings.updateBooking(this.state.booking)
+    .then(result => {
+      this.setLoading(false);
+      this.props.onSaved(this.state.booking);
+    })
+    .catch(e => {
+      this.setLoading(false);
+      this.props.onError();
+      console.log(e)
+    });
+  }
+
+  saveButton(isDisabled) {
+    return isDisabled
+      ? <Button color="primary" disabled>Save</Button> 
+      : <Button color="primary" onClick={(e) => this.submit(e)}>Save</Button>;
+
+  }
+
+  setLoading(isLoading) {
+    this.setState({isLoading: isLoading})
+  }
+
+  handleStartDateChange(date)
+  {
+    //let field = evt.target.id;
+    //let value = evt.target.value
+    this.setState({ tmpStartsAt: date });
+    this.setState({
+      booking: { ...this.state.booking, ['startsAt']: date },
+      isTouched: { ...this.state.isTouched, ['startsAt']: true },
+    });
+  }
+
+  handleEndDateChange(date)
+  {
+    //let field = evt.target.id;
+    //let value = evt.target.value
+    this.setState({ tmpEndsAt: date });
+    this.setState({
+      booking: { ...this.state.booking, ['endsAt']: date },
+      isTouched: { ...this.state.isTouched, ['endsAt']: true },
+    });
+  }
+
+  renderDateFormGroup(field, time) {
+    const placeholder = this.labels[field];
+    return (
+      <FormGroup>
+        {this.renderLabel(field, field)}
+        <DatePicker
+                selected={moment(this.state.booking[field])}
+                onChange={(field=='startsAt') ? this.handleStartDateChange.bind(this) : this.handleEndDateChange.bind(this)}
```

Comment:
could be `onChange={(e) => this.handleDateChange(field, e)}` if you consolidated the 2 methods into 1